File: ex08/matrix.cpp
#include <iostream>
#include <vector>

int main() {
    int row, col;

    std::cout << "행과 열을 입력하세요 (row, col) : ";
    std::cin >> row >> col;

    std::vector<std::vector<int>> first_matrix(row, std::vector<int>(col));
    std::vector<std::vector<int>> second_matrix(row, std::vector<int>(col));

    std::cout << "첫 번째 행렬 : \n";
    for(int i=0; i<row; i++){
        for(int j=0; j<col; j++){
            std::cin >> first_matrix[i][j];
        }
    }

    std::cout << "두 번째 행렬 : \n";
    for(int i=0; i<row; i++){
        for(int j=0; j<col; j++){
            std::cin >> second_matrix[i][j];
        }
    }

    std::cout << "두 행렬을 더한 결과 : \n";
     for(int i=0; i<row; i++){
        for(int j=0;j<col;j++) {
            printf("%d ", first_matrix[i][j] + second_matrix[i][j]);
        }
         printf("\n");
     }

     return 0;
}

File: ex09/matrix.rs
use std::io::{self, Write, stdin};

fn main() -> io::Result<()> {
    let mut buffer = String::new();

    print!("행과 열을 입력하세요 (row,col): ");
    io::stdout().flush()?;

    stdin().read_line(&mut buffer)?;

    let vec: Vec<usize> = buffer.trim().split_whitespace()
                                .map(|x| x.parse::<usize>().unwrap())
                                .collect();

    let (row, col) = (vec[0], vec[1]);

    let mut first_matrix: Vec<Vec<i32>> = vec![vec![0; col]; row];
    let mut second_matrix: Vec<Vec<i32>> = vec![vec![0; col]; row];

   println!("첫 번째 행렬:");
   for i in 0..row {
       let mut input_string = String::new();
       stdin().read_line(&mut input_string)?;
       first_matrix[i] = input_string.trim().split_whitespace()
                                      .map(|x| x.parse::<i32>().unwrap())
                                      .collect();
   }

   println!("두 번째 행렬:");
   for i in 0..row {
       let mut input_string = String::new();
       stdin().read_line(&mut input_string)?;
       second_matrix[i] = input_string.trim().split_whitespace()
                                       .map(|x| x.parse::<i32>().unwrap())
                                       .collect();
   }

   println!("두 행렬을 더한 결과:");
   
   for i in 0..row {
       for j in 0..col {
           print!("{} ", first_matrix[i][j] + second_matrix[i][j]);
       }
       println!("");
   }

   Ok(())
}
